File: v-poc/sdk/GrowingIO/collect.spring/src/main/java/org/example/collect/utils/JsonUtils.java
package org.example.collect.utils;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.SneakyThrows;
import lombok.experimental.UtilityClass;

@UtilityClass
public class JsonUtils{

    private final ObjectMapper objectMapper = new ObjectMapper();

    @SneakyThrows
    public <T> T  toObject(String json, Class<T> clazz) {
        return objectMapper.readValue(json, clazz);
    }

    @SneakyThrows
    public <T> T toObject(String json, TypeReference<T> typeRef) {
        return objectMapper.readValue(json, typeRef);
    }

    @SneakyThrows
    public String toJson(Object obj) {
        return objectMapper.writeValueAsString(obj);
    }
}

File: a-demo/protect-api/src/main/kotlin/org/example/util/JsonUtils.kt
package org.example.util

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper


object JsonUtils {
	
	private val objectMapper = ObjectMapper()
	
	fun <T> json2Object(json: String, clazz: Class<T>): T = objectMapper.readValue(json, clazz)
	
	fun <T> jsonToObject(json: String?, typeRef: TypeReference<T>): T = objectMapper.readValue(json, typeRef)
	
	fun object2Json(obj: Any): String = objectMapper.writeValueAsString(obj)
}